students/wonjae/users/views.py

```diff
@@ -1,3 +1,22 @@
-from django.shortcuts import render
+import json
+import re
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+
+from users.models import User
+
+class UserView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            if not re.match(r'^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$', data['email']):
+                return JsonResponse({"message" : "Email Invalid !!!"}, status=400)
+            if not re.match(r'^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$', data['password']):
+            	return JsonResponse({"message" : "Password Invalid !!!"}, status=400)
+            if User.objects.filter(email=data['email']).exists():
+                return JsonResponse({"message" : "Email Overlapped !!!"}, status=400)
+            User.objects.create(name=data['name'], email=data['email'], password=data['password'], telephone=data['telephone'])
+            return JsonResponse({"message" : "SUCCESS"}, status=201)
+        except KeyError:
+            return JsonResponse({"message" : "Key Error !!!"}, status=400)
```

Comment:
피드백 1. 정규 표현식은 다른 email을 검증하는 곳에서 사용될 수 있으므로, 밖으로 빼주세요-!
```python
EMAIL_REGEX = 'r'^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
```

피드백 2. 코드 사이에 간격이 없어서 가독성이 좋지 않습니다. 
```suggestion
            if not re.match(EMAIL_REGEX, data['email']):
                return JsonResponse({"message" : "Email Invalid !!!"}, status=400)
                
            if not re.match(r'^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$', data['password']):
            	return JsonResponse({"message" : "Password Invalid !!!"}, status=400)
            	
            if User.objects.filter(email=data['email']).exists():
                return JsonResponse({"message" : "Email Overlapped !!!"}, status=400)
                
            User.objects.create(name=data['name'], email=data['email'], password=data['password'], telephone=data['telephone'])
            
            return JsonResponse({"message" : "SUCCESS"}, status=201)
            
        except KeyError:
            return JsonResponse({"message" : "Key Error !!!"}, status=400)
```